### PR TITLE
Remove deprecated WithEvents component

### PR DESCRIPTION
**Pull request by shilman**
Issue: #3629 

## What I did

Removed all references to WithEvents from addon-events and the examples

## How to test

Is this testable with Jest or Chromatic screenshots?
Does this need a new example in the kitchen sink apps?
Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by Hypnosphi** [comment]
I've introduced this deprecation only in 4.0-alpha. Are we sure we don't want to wait some more?

**Comment by shilman** [comment]
@Hypnosphi I'm fine either way. addon-events isn't used much (4k/mo npm downloads), so I think it's probably fine to remove it. Can you take a look at the remaining deprecations in #3629 (knobs, backgrounds) and comment on those? I think the knobs deprecations are also introduced in 4.0-alpha, and those are used a lot so maybe we should delay. LMK in the issue.

**Review by z4o4z — APPROVED** [review]

File: scripts/build-storybooks.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (ndelangen) [inline scripts/build-storybooks.sh]: This seems unrelated?

**Review by danielduan — APPROVED** [review]
